pkg/api/stars.go

```diff
@@ -6,9 +6,37 @@ import (
 
 	"github.com/grafana/grafana/pkg/api/response"
 	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
 	"github.com/grafana/grafana/pkg/web"
 )
 
+func (hs *HTTPServer) GetStars(c *models.ReqContext) response.Response {
+	// SELECT uid FROM star JOIN dashboard ON star.dashboard_id = dashboard.id
+	// WHERE star.user_id = 1
+	type row struct {
+		Uid string
+	}
+	rows := make([]*row, 0, 1000)
+
+	err := hs.SQLStore.WithDbSession(c.Req.Context(), func(sess *sqlstore.DBSession) error {
```

Comment:
I am not sure what the preferred style is here -- ideally we do not have raw SQL directly in the HTTP server!!!  This endpoint could be refactored when it is backed by an independent service (that ideally can "star" items that do not have internal dashboar_ids)

Reply:
I found this sqlstore method GetUserStars is doing quite similar thing just instead of UID, it is using ID. Since this method is only used internally (not called directly by any endpoint) we can modify the method and use it directly instead of query DB from API.